packages/data-service/src/csfle-collection-tracker.spec.ts

```diff
@@ -33,8 +26,22 @@ describe('CSFLECollectionTracker', function () {
   });
 
   async function createTracker(
-    autoEncryption: AutoEncryptionOptions = {}
-  ): Promise<[CSFLECollectionTracker, DataService]> {
+    autoEncryption:
+      | AutoEncryptionOptions
+      | ((keys: [Binary, Binary]) => AutoEncryptionOptions) = {}
+  ): Promise<[CSFLECollectionTracker, DataService, Binary, Binary]> {
+    if (typeof autoEncryption === 'function') {
+      // This autoEncryption config depends on the keys, so we first need
+      // to set up the keys with one DataService instance,
+      // then create the 'real' DataService instance.
+      const [, tempDataService, someKey1, someKey2] = await createTracker({});
+      await tempDataService.disconnect();
+      const [tracker, dataService] = await createTracker(
+        autoEncryption([someKey1, someKey2])
+      );
+      return [tracker, dataService, someKey1, someKey2];
+    }
```

Comment:
The updates for this file were quite a bit larger than I anticipated, unfortunately.

Most of this is because libmongocrypt has become a bit smarter about collection management, including `createCollection` calls, and that messes with the expectations in this test.